server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/LocalAlignedGroupByExecutor.java

```diff
@@ -0,0 +1,388 @@
+    // update filter by TTL
+    timeFilter = queryDataSource.updateFilterUsingTTL(timeFilter);
+
+    // init AlignedSeriesAggregateReader for aligned series
+    Set<String> allSensors = new HashSet<>(((AlignedPath) path).getMeasurementList());
+    reader =
+        new AlignedSeriesAggregateReader(
+            (AlignedPath) path,
+            allSensors,
+            TSDataType.VECTOR,
+            context,
+            queryDataSource,
+            timeFilter,
+            null,
+            fileFilter,
+            ascending);
+
+    preCachedData = null;
+    timeRange = new TimeRange(Long.MIN_VALUE, Long.MAX_VALUE);
+    lastReadCurArrayIndex = 0;
+    lastReadCurListIndex = 0;
+    this.ascending = ascending;
+  }
+
+  @Override
+  public void addAggregateResult(AggregateResult aggrResult) {
+    throw new UnsupportedOperationException(
+        "This method is not supported in LocalAlignedGroupByExecutor");
+  }
+
+  @Override
+  public List<AggregateResult> calcResult(long curStartTime, long curEndTime)
+      throws IOException, QueryProcessException {
+    throw new UnsupportedOperationException(
+        "This method is not supported in LocalAlignedGroupByExecutor");
+  }
+
+  @Override
+  public Pair<Long, Object> peekNextNotNullValue(long nextStartTime, long nextEndTime)
+      throws IOException {
+    throw new UnsupportedOperationException(
+        "This method is not supported in LocalAlignedGroupByExecutor");
+  }
+
+  public void addAggregateResultList(List<AggregateResult> aggrResultList) {
+    results.add(aggrResultList);
+  }
+
+  private boolean isEndCalc() {
+    for (List<AggregateResult> resultsOfOneMeasurement : results) {
+      for (AggregateResult result : resultsOfOneMeasurement) {
+        if (!result.hasFinalResult()) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  public List<List<AggregateResult>> calcAlignedResult(long curStartTime, long curEndTime)
+      throws IOException, QueryProcessException {
+
+    // clear result cache
+    for (List<AggregateResult> resultsOfOneMeasurement : results) {
+      for (AggregateResult result : resultsOfOneMeasurement) {
+        result.reset();
+      }
+    }
+
+    timeRange.set(curStartTime, curEndTime - 1);
+    if (calcFromCacheData(curStartTime, curEndTime)) {
+      return results;
+    }
+
+    // read page data firstly
+    if (readAndCalcFromPage(curStartTime, curEndTime)) {
+      return results;
+    }
+
+    // read chunk data secondly
+    if (readAndCalcFromChunk(curStartTime, curEndTime)) {
+      return results;
+    }
+
+    // read from file
+    while (reader.hasNextFile()) {
+      // try to calc from fileMetaData
+      Statistics fileStatistics = reader.currentFileStatistics();
+      if (fileStatistics != null && !isEmpty(fileStatistics)) {
+        if (fileStatistics.getStartTime() >= curEndTime) {
+          return results;
+        }
+        if (reader.canUseCurrentFileStatistics()
```

Comment:
The `canUseCurrentFileStatistics()` function in `AlignedSeriesAggregateReader` has the same wrong thing like the above, it use the first sub sensor's statistics instead of time column's. You can change that by the way.

Reply:
fixed